R/data-combined.R

```diff
@@ -346,6 +349,144 @@ DataCombined <- R6::R6Class(
       invisible(self)
     },
 
+    #' @description
+    #'
+    #' Get a dataframe for Predicted versus Observed plot
+    #'
+    #' @details
+    #'
+    #' The `threshold` argument can make a big difference to the returned
+    #' dataframe. This argument specifies the tolerance for comparing equality
+    #' of observed and simulated time values (usually found under `xValues`
+    #' column).
+    #'
+    #' Specifically, there are two possibilities to consider. To illustrate
+    #' them, let's say the observed `xValue` is `2` and the simulated values are
+    #' `c(1.4, 2.2, 3.8, 4.0)`.
+    #'
+    #' If the threshold is too **liberal**, multiple simulated values will be
+    #' selected. In such cases, only the first of these is retained. For
+    #' example, with `threshold = 1`, two simulated values (`c(1.4, 2.2`) are
+    #' selected and only the first one (`1.4`) is retained.
+    #'
+    #' On the other hand, if the threshold is too **stringent**, no simulated
+    #' values might be selected. With a `threshold = 0.1`, no simulated values
+    #' will be selected for out example data.
+    #'
+    #' @param threshold Absolute numerical distance by which the closest
+    #'   `xValue` (time, i.e.) in the observed data may differ from `xValue` in
+    #'   simulated data to be accepted as *equal*. If no threshold is specified
+    #'   (default), then the square root of smallest positive floating-point
+    #'   number on your machine will be used (`.Machine$double.eps^0.5`) as a
+    #'   threshold. For all intents and purposes, this amounts to an **exact**
+    #'   match between two values being compared.
+    toObsVsPredDataFrame = function(threshold = NULL) {
+      validateIsNumeric(threshold, nullAllowed = TRUE)
+
+      # We don't want this function to tamper with combined dataframe, so
+      # make an internal copy for this function scope.
+      df <- private$.dataCombined %>%
+        # Remove unnecessary columns
+        private$.cleanDF(.) %>%
+        # Datasets which are not part of any grouping can't be paired
+        dplyr::filter(!is.na(group))
+
+      if (is.null(df)) {
+        stop(
+          "There are currently no datasets. You can add them with `$addDataSets()` and/or `$addSimulationResults()` methods.",
```

Comment:
Should not we move all those strings into a constant file with messages as we do with everything else?

Reply:
Yes, definitely! 

Should I move them to _ospsuite.utils_?

Reply:
I don;t think so. Those messages should leave in the package are they are very specific to what ospsuite-r is doing